contracts/integrations/test/examples/simple.ts

```diff
@@ -0,0 +1,75 @@
+import { blockchainTests, expect } from '@0x/contracts-test-utils';
+import { BigNumber } from '@0x/utils';
+
+import { artifacts, TestFrameworkContract } from '../../src';
+import { FunctionAssertion, Result } from '../utils/function_assertions';
+
+// These tests provide examples for how to use the "FunctionAssertion" class to write
+// tests for "payable" and "nonpayable" Solidity functions as well as "pure" and "view" functions.
+blockchainTests('TestFramework', env => {
+    let exampleContract: TestFrameworkContract;
+
+    before(async () => {
+        exampleContract = await TestFrameworkContract.deployFrom0xArtifactAsync(
+            artifacts.TestFramework,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    describe('numberSideEffect', () => {
+        let assertion: FunctionAssertion;
+
+        before(async () => {
+            const condition = {
+                before: async () => {},
+                after: async (beforeInfo: any, result: Result) => {
+                    // Ensure that the call was successful.
+                    expect(result.success).to.be.true();
+
+                    // Ensure that the correct counter was returned.
+                    const counter = await exampleContract.counter.callAsync();
+                    expect(result.data).bignumber.to.be.eq(counter);
+                },
+            };
+            assertion = new FunctionAssertion(exampleContract.numberSideEffect, condition);
+        });
+
+        it('should return the correct counter', async () => {
+            await assertion.runAsync();
```

Comment:
So normally I hate this pattern because it isn't very explicit, but because we define the assertions within the `describe` block it isn't so bad, so :+1: